databroker/src/broker.rs

```diff
@@ -1886,39 +1892,60 @@ impl<'a, 'b> AuthorizedAccess<'a, 'b> {
         }
     }
 
-    async fn validate_actuator_value(
+    async fn validate_actuator_update(
         &self,
         vss_id: &i32,
         data_value: &DataValue,
     ) -> Result<(), (ActuationError, String)> {
         let result_entry = self.get_entry_by_id(*vss_id).await;
         match result_entry {
             Ok(entry) => {
-                let validation = entry.validate_value(data_value);
-                let vss_path = entry.metadata.path;
+                let metadata = entry.metadata.clone();
+                let vss_path = metadata.path;
+                if metadata.entry_type != EntryType::Actuator {
+                    let message = format!("Tried to set a value for a non-actuator: {}", vss_path);
+                    return Err((ActuationError::WrongType, message));
+                }
+                let validation = entry.validate_actuator_value(data_value);
                 match validation {
                     Ok(_) => Ok(()),
                     Err(UpdateError::OutOfBounds) => {
-                        let message = format!(
-                            "Out of bounds value provided for {}: {} | Expected range [min: {}, max: {}]",
-                            vss_path,
-                            data_value,
-                            entry.metadata.min.map_or("None".to_string(), |value| value.to_string()),
-                            entry.metadata.max.map_or("None".to_string(), |value| value.to_string()),
-                        );
-                        Err((ActuationError::OutOfBounds, message))
+                        let mut message = "No message defined".to_string();
```

Comment:
I am not really sure if we should try to be smart here and come up with the reason why we got "OutOfBounds". We do not only report OutOfBounds when min/max or allowed is exceeded,also in some other cases like when the VSS datatype is uint8 but Broker got a 275 as value.

If needed it would be better I think to split UpdateError::OutOfBounds to multiple ones indicating reason, like UpdateError::OutOfBoundsAllowed, UpdateError::OutOfBoundsMinMax, UpdateError::OutOfBoundsType

Or alternatively create a message deepr down in the stack

Reply:
I prefer the approach of extending UpdateError and propagating the correct type of error upstream.
Thanks.